user_gen.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+
+
+def simp_poisson_process(rate=1, size=None):
+    sum_exp_draws = np.cumsum(np.random.exponential(scale=rate, size=size))
+    # sum_exp_draws = [sum(exp_draws[:i]) for i, x in enumerate(exp_draws)]
+    return sum_exp_draws
+
+
+def poisson_plus(lam=1, plus=1, size=None):
+    return np.random.poisson(lam=lam, size=size) + plus
+
+
+def int_exponential(rate=1, size=None):
+    exp_draws = np.random.exponential(scale=rate, size=size)
+    return np.round(exp_draws)
+
+
+def diffs_btw_exp(rate=1, size=None):
+    sorted_draws = sorted(np.random.exponential(scale=rate, size=size))
+    seq_diffs = [x-sorted_draws[i-1] for i, x in enumerate(sorted_draws)]
+    seq_diffs[0] = sorted_draws[0]
+    return np.array(seq_diffs)
+
+
+def is_pos_inc_seq(seq):
+    return (sorted(seq) == list(seq) and all([x >= 0 for x in seq]))
+
+
+def is_any_int(x):
+    return isinstance(x, (int, np.int64))
+
+
+def is_pos_int(seq):
+    return (all([(x > 0 and is_any_int(x)) for x in seq]))
+
+
+class Simple_Job_Set(object):
+    def __init__(self, num_jobs):
+        self.num_jobs = num_jobs
+
+    @property
+    def users_per_job(self):
+        return self._users_per_job
+
+    @users_per_job.setter
+    def users_per_job(self, users_per_job):
+        if is_pos_int(users_per_job):
+            self._users_per_job = users_per_job
+        else:
+            raise ValueError("All jobs must be assigned at least one user")
+
+    @property
+    def start_times(self):
+        return self._start_times
+
+    @start_times.setter
+    def start_times(self, start_times):
+        if is_pos_inc_seq(start_times):
+            self._start_times = start_times
+        else:
+            raise ValueError("start times must be positive and increasing")
+
+    def rand_num_user_gen(self, params=None, func=None):
+        """
+        generates an integer number of users
+        params is a dict of parameters
+        func is a function with a signature like numpy.random functions
+        """
+        if params is None:
+            params = {"lam": 5,
+                      "plus": 1
+                      }
+        if func is None:
+            func = poisson_plus
+        self.users_per_job = func(**params, size=self.num_jobs)
+
+    def rand_start_times_gen(self, params=None, func=None):
+        """
+        params is a dict of parameters
+        func is a function with a signature like numpy.random functions
+        """
+        if params is None:
+            params = {"rate": 5}
+        if func is None:
+            func = simp_poisson_process
+        self.start_times = func(**params, size=self.num_jobs)
+
+    def make_job_set(self, job_type=None):
```

Comment:
also feels like the job_type could be created upon `init` of this class instead of in this method